official/utils/misc/distribution_utils.py

```diff
@@ -23,43 +23,51 @@
+      Accepted values are 'off', 'default', 'one_device', 'mirrored',
+      'parameter_server', 'collective', case insensitive. 'off' means not to use
+      Distribution Strategy; 'default' means to choose from `MirroredStrategy`
+      or `OneDeviceStrategy` according to the number of GPUs."
     num_gpus: Number of GPUs to run this model.
-    all_reduce_alg: Specify which algorithm to use when performing all-reduce.
-      See tf.contrib.distribute.AllReduceCrossDeviceOps for available
-      algorithms. If None, DistributionStrategy will choose based on device
-      topology.
-    turn_off_distribution_strategy: when set to True, do not use any
-      distribution strategy. Note that when it is True, and num_gpus is
-      larger than 1, it will raise a ValueError.
+    all_reduce_alg: Optional. Specify which algorithm to use when performing
+      all-reduce. See tf.contrib.distribute.AllReduceCrossDeviceOps for
+      available algorithms. If None, DistributionStrategy will choose based on
+      device topology.
 
   Returns:
-    tf.contrib.distribute.DistibutionStrategy object.
+    tf.distribute.DistibutionStrategy object.
   Raises:
-    ValueError: if turn_off_distribution_strategy is True and num_gpus is
-    larger than 1
+    ValueError: if `distribution_strategy` is 'off' or 'one_device' and
+      `num_gpus` is larger than 1; or `num_gpus` is negative.
   """
-  if num_gpus == 0:
-    if turn_off_distribution_strategy:
-      return None
+  if num_gpus < 0:
+    raise ValueError("`num_gpus` can not be negative.")
+
+  distribution_strategy = distribution_strategy.lower()
+  if distribution_strategy == "off":
+    if num_gpus > 1:
+      raise ValueError("When {} GPUs are specified, distribution_strategy flag "
+                       "cannot be set to 'off'.".format(num_gpus))
+    return None
+
+  if (distribution_strategy == "one_device" or
+      (distribution_strategy == "default" and num_gpus <= 1)):
+    if num_gpus == 0:
+      return tf.contrib.distribute.OneDeviceStrategy("device:CPU:0")
     else:
-      return tf.contrib.distribute.OneDeviceStrategy('device:CPU:0')
-  elif num_gpus == 1:
-    if turn_off_distribution_strategy:
-      return None
-    else:
-      return tf.contrib.distribute.OneDeviceStrategy('device:GPU:0')
-  elif turn_off_distribution_strategy:
-    raise ValueError('When {} GPUs are specified, '
-                     'turn_off_distribution_strategy flag cannot be set to'
-                     'True.'.format(num_gpus))
-  else:  # num_gpus > 1 and not turn_off_distribution_strategy
-    devices = ['device:GPU:%d' % i for i in range(num_gpus)]
+      if num_gpus > 1:
+        raise ValueError("`OneDeviceStrategy` can not be used for more than "
+                         "one device.")
+      return tf.contrib.distribute.OneDeviceStrategy("device:GPU:0")
+
+  if distribution_strategy in ("mirrored", "default"):
+    devices = ["device:GPU:%d" % i for i in range(num_gpus)]
```

Comment:
the check i was asking for is actually this case: distribution_strategy=mirrored, and num_gpus=0. 
We will pass empty list to MirroredStrategy. what happens in that case? (not a big deal, just checking)

Reply:
I thought MS would use cpu but turns out not. So updated the code to cover gpus=0 case.